### PR TITLE
SBERDOMA-872 update and configure jest

### PR DESCRIPTION
**Pull request by MrFoxPro**

### CODE REVIEW

File: apps/condo/jest.config.js [hunk 1]
```diff
@@ -1,4 +1,21 @@
...
 module.exports = {
+    // this is not working in jest =( https://github.com/facebook/jest/issues/11500
+    // testTimeout: 1000000,
+    silent: true,   
+    // https://stackoverflow.com/questions/43864793/why-does-jest-runinband-speed-up-tests
+    maxWorkers: conf.IN_CI ? 1 : 4, 
+    reporters: [
+        'default',
+        [
+            'jest-stare',
```
Comment (AntonAL) [inline apps/condo/jest.config.js:13]: Could you please create a documentation file `docs/testing.md` and describe the usage of this plugin and the availability of `test-stare` directory for published applications in our servers.
PS: Do we really need this extra visuals?

**Review by AntonAL — COMMENTED** [review]
In the task it states to unclutter console logging.
I don't see, what was changed, — the same ~1500 lines of test logs in this and other pull-requests.
Please, show the differences, how. this extra plugins affected test logs.

File: apps/condo/jest.config.js (hunk 1 above)
Comment (MrFoxPro) [reply apps/condo/jest.config.js:17]: I've created custom reporter to hide this logs.
 https://github.com/MrFoxPro/jest-clean-reporter

**Comment by MrFoxPro** [comment]
https://github.com/MrFoxPro/jest-clean-reporter

**Comment by MrFoxPro** [comment]
We need to check:

- [ ] If tests passes, CI end up working with success status
- [ ] If tests fails, CI end up working with error

**Review by Dimitreee — APPROVED** [review]

**Comment by MrFoxPro** [comment]
https://github.com/facebook/jest/issues/11463

**Review by leonid-d — APPROVED** [review]

File: apps/condo/domains/notification/schema/SendMessageService.test.js [hunk 2]
```diff
@@ -8,7 +8,8 @@ const { MESSAGE_SENDING_STATUS, MESSAGE_RESENDING_STATUS, MESSAGE_DELIVERED_STAT
 const sleep = async (time) => (new Promise((resolve => {
     setTimeout(resolve, time)
 })))
-
+// https://github.com/facebook/jest/issues/11500
+jest.setTimeout(30000)
```
Comment (pahaz) [inline apps/condo/domains/notification/schema/SendMessageService.test.js:12]: it's better to move it inside https://github.com/open-condo-software/condo/blob/master/apps/condo/jest.setup.js

File: .github/workflows/nodejs.condo.test.yml [hunk 3]
```diff
@@ -84,3 +84,4 @@ jobs:
           DATABASE_URL: ${{ matrix.database }}
           NODE_ENV: development
           DISABLE_LOGGING: true
+          IN_CI: true
```
Comment (pahaz) [inline .github/workflows/nodejs.condo.test.yml:87]: it looks useless.

File: apps/condo/jest.config.js [hunk 4]
```diff
@@ -1,4 +1,34 @@
+const conf = require('@core/config')
+
+/** @type {import('@jest/types').Config.InitialOptions} */
 module.exports = {
+    // this is not working in jest =( https://github.com/facebook/jest/issues/11500
+    // testTimeout: 1000000,
+    silent: true,
+    verbose: false,
+    // https://stackoverflow.com/questions/43864793/why-does-jest-runinband-speed-up-tests
+    maxWorkers: conf.IN_CI ? 1 : 4,
```
Comment (pahaz) [inline apps/condo/jest.config.js:10]: it's better to fix the tests to support parallel mode

File: apps/condo/package.json [hunk 5]
```diff
@@ -97,7 +97,9 @@
     "@types/react": "^17.0.1",
     "babel-jest": "^26.6.3",
     "identity-obj-proxy": "^3.0.0",
-    "jest": "^26.6.3",
+    "jest": "27.0.6",
+    "jest-clean-reporter": "https://github.com/MrFoxPro/jest-clean-reporter",
```
Comment (pahaz) [inline apps/condo/package.json:101]: it looks super starnge

**Review by pahaz — CHANGES_REQUESTED** [review]

File: apps/condo/jest.setup.js [hunk 6]
```diff
@@ -14,5 +16,13 @@ beforeAll(async () => {
 })
 
 afterAll(async () => {
+    if (redisGuard.db) {
```
Comment (pahaz) [inline apps/condo/jest.setup.js:19]: it doesn't work =/

File: apps/condo/jest.config.js [hunk 7]
```diff
@@ -1,4 +1,34 @@
...
+    noStackTrace: true,
```
Comment (pahaz) [inline apps/condo/jest.config.js:11]: is it ok to disable stack trace for all developers?

File: packages/@core.keystone/setup.utils.js [hunk 8]
```diff
@@ -46,17 +46,18 @@ function getAdapter (databaseUrl) {
...
+        adapter.connect = () => { throw new Error('UndefinedAdapter.connect() call!') }
+        adapter.postConnect = () => { throw new Error('UndefinedAdapter.postConnect() call!') }
+        adapter.checkDatabaseVersion = () => { throw new Error('UndefinedAdapter.checkDatabaseVersion() call!') }
         return adapter
     } else {
         throw new Error(`getAdapter() call with unknown schema: ${databaseUrl}`)
     }
 }
-
+let redisClientHandler = {}
```
Comment (pahaz) [inline packages/@core.keystone/setup.utils.js:57]: it looks like hack!

Comment (MrFoxPro) [reply packages/@core.keystone/setup.utils.js:57]: It is

**Comment by pahaz** [comment]
Why you don't use `DISABLE_LOGGING` ?

**Pull request closed without merging**